src/com/griddynamics/cinema/CinemaCli.java

```diff
@@ -0,0 +1,134 @@
+            row = scanner.nextInt();
+            System.out.print("Enter a seat number in that row:\n> ");
+            seat = scanner.nextInt();
+            
+            errorMessage = row > amountOfRows || seat > amountOfSeatsInEachRow ? "Wrong input!\n"
+                    : getCinema().getHall().getRows().get(row - 1).getSeatsInEachRow().get(seat - 1).isReserved()
+                            ? "That ticket has already been purchased!\n" : "";
+            
+            System.out.print('\n' + errorMessage);
+        } while (!errorMessage.isEmpty());
+        
+        cinema.getHall().getRows().get(row - 1).getSeatsInEachRow().get(seat - 1).reserve();
+        System.out.printf("Ticket price: $%d\n", getTicketPrice(row));
+    }
+    
+    private void statistics() {
+        System.out.println("\nNumber of purchased tickets: " + getPurchasedTicketsAmount());
+        System.out.printf("Percentage: %.2f%%\n", getPercentageOfPurchasedTickets());
+        System.out.println("Current income: $" + getCurrentIncome());
+        System.out.println("Total income: $" + getTotalIncome());
+    }
+    
+    private void showSeats() {
+        System.out.println("\nCinema:\n" + getCinema());
+    }
+    
+    private int getTicketPrice(final int row) {
+        return isLargeCinemaHall && row > amountOfRowsWithFullPriceSeats ? TICKET_DISCOUNT_PRICE : TICKET_FULL_PRICE;
+    }
+    
+    private int getPurchasedTicketsAmount() {
+        return (int)getCinema().getHall()
+                               .getRows()
+                               .stream()
+                               .mapToLong(row -> row.getSeatsInEachRow().stream().filter(Seat::isReserved).count())
+                               .sum();
+    }
+    
+    private double getPercentageOfPurchasedTickets() {
+        return (double)getPurchasedTicketsAmount() / totalSeatsAmount * ONE_HUNDRED_PERCENT;
+    }
+    
+    private int getCurrentIncome() {
+        int purchasedDiscountTicketsAmount = !isLargeCinemaHall ? 0 : (int)getCinema().getHall()
+                                                                                      .getRows()
+                                                                                      .stream()
+                                                                                      .skip(amountOfRowsWithFullPriceSeats)
+                                                                                      .mapToLong(row -> row.getSeatsInEachRow()
+                                                                                                           .stream()
+                                                                                                           .filter(Seat::isReserved)
+                                                                                                           .count())
+                                                                                      .sum();
```

Comment:
That style is great until we don't use Command + Alt + L.
Do we really need to create stream chain methods in that way?

Reply:
Actually we don't. I will change it to more readable format. Thank you!